File: app/models/support_message.py
"""
Minimal Support Message model for MVP
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from enum import Enum


class SenderRole(str, Enum):
    USER = "user"
    ADMIN = "admin"


class SupportMessage(BaseModel):
    """
    Minimal Message Model
    Plain text only. No attachments.
    """
    ticket_id: str = Field(..., description="Parent Ticket ID")
    sender_id: str = Field(..., description="ID of the sender")
    sender_role: SenderRole = Field(..., description="Role of sender: user/admin")
    content: str = Field(..., min_length=1, max_length=4000, description="Plain text content")
    created_at: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        use_enum_values = True

File: models/SupportMessage.js
const mongoose = require('mongoose');

const SenderRole = {
  USER: 'user',
  ADMIN: 'admin'
};

const supportMessageSchema = new mongoose.Schema({
  ticket_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Ticket',
    required: true
  },
  sender_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User', // Or Admin if you have separate collection, but usually User for both or generic ID
    required: true
  },
  sender_role: {
    type: String,
    enum: Object.values(SenderRole),
    required: true
  },
  content: {
    type: String,
    required: true,
    minlength: 1,
    maxlength: 4000
  }
}, {
  timestamps: { createdAt: 'created_at', updatedAt: false } // Only createdAt
});

supportMessageSchema.index({ ticket_id: 1, created_at: 1 });

const SupportMessage = mongoose.model('SupportMessage', supportMessageSchema);

module.exports = {
  SupportMessage,
  SenderRole
};
